### PR TITLE
Fix GRALLOC1_PFN_LOCK_FLEX multithread issue.

### PR DESCRIPTION
**Pull request by dulayev**
lockFlex fills outFlexLayout->planes with pointer to global variable,
doesn't work from multiple threads.
Solution: rely that caller allocates needed planes structures,
just use them for output.

gralloc1.h about GRALLOC1_PFN_GET_NUM_FLEX_PLANES:
This may be used to efficiently allocate only as many plane structures as necessary before calling into lockFlex.
It means that lockFlex caller is responsible for outFlexLayout->planes allocation.

It also demand fix in GRALLOC1_PFN_GET_NUM_FLEX_PLANES:
should return 3 for HAL_PIXEL_FORMAT_NV12_TILED_INTEL.

Jira: None
Test: check memory regions mapped from different threads don't overlap

### CODE REVIEW

File: cros_gralloc/gralloc1/cros_gralloc1_module.cc [hunk 1]
```diff
@@ -539,7 +538,11 @@ int32_t CrosGralloc1::getNumFlexPlanes(buffer_handle_t buffer, uint32_t *outNumP
 		return CROS_GRALLOC_ERROR_BAD_HANDLE;
 	}
 
-	*outNumPlanes = drv_num_planes_from_format(hnd->format);
+	if (DRM_FORMAT_NV12_Y_TILED_INTEL == hnd->format) {
+		*outNumPlanes = 3;
```
Comment (tpalli) [inline cros_gralloc/gralloc1/cros_gralloc1_module.cc]: Shouldn't you rather fix drv_num_planes_from_format to return correct value for this format? Also, I think this change should be in a separate patch as it is a behavioural change.

Comment (dulayev) [reply cros_gralloc/gralloc1/cros_gralloc1_module.cc]: I don't think so.
drv_num_planes_from_format is used to fill struct bo->num_planes, which is widely used in minigbm.
bo->num_planes treats interleaved planes (like UV) as one plane and I don't want to change this.
getNumFlexPlanes treats interleaved planes as separate planes, so it doesn't match drv_num_planes_from_format.

But I agree that CrosGralloc1::getNumFlexPlanes could be improved: 
logic should be similar to CrosGralloc1::lockYCbCr: check if format is supported YUV,
and return always 3, don't use drv_num_planes_from_format at all.

Comment (tpalli) [reply cros_gralloc/gralloc1/cros_gralloc1_module.cc]: Ah ok, so it's is 'logical amount' of planes in this case, that explains!

**Comment by js0701** [comment]
closed as merged
